src/cpp/rtps/writer/StatelessWriter.cpp

```diff
@@ -377,7 +377,7 @@ bool StatelessWriter::change_removed_by_history(
     }
 
     const uint64_t sequence_number = change->sequenceNumber.to64long();
-    if (sequence_number > last_sequence_number_sent_)
+    if (sequence_number <= last_sequence_number_sent_)
```

Comment:
This can be reverted. We only need to notify the condition if we are removing a sequence number that is greater than the last one sent. In that case, a thread blocked inside `wait_for_acknowledgement` should wake-up and return due to the new condition (sequence not present in history)

```suggestion
    if (sequence_number > last_sequence_number_sent_)
```